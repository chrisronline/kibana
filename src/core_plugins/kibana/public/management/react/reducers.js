/* eslint-disable */
import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux'

import app from './app/app.reducer';
import indexPattern from './index-pattern/index-pattern.reducer';

export const rootReducer = combineReducers({
  routing: routerReducer,
  app,
  indexPattern,
});
