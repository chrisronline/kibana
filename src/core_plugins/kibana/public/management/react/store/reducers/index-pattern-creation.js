import { handleActions } from 'redux-actions';
import { createSelector } from 'reselect';
import { chunk, sortBy as sortByLodash } from 'lodash';

import {
  selectTimeField,
  fetchedTimeFields,
  goToNextPage,
  goToPreviousPage,
  changeSort,
  fetchedIndices,
  includeSystemIndices,
  excludeSystemIndices,
  creatingIndexPattern,
  createdIndexPattern,
} from '../actions/index-pattern-creation';

import {
  getIndexPatternCreate,
} from '../../reducers';

const defaultState = {
  isIncludingSystemIndices: false,
  isCreating: false,
  timeFields: {
    timeFields: undefined,
    selectedTimeField: undefined,
  },
  results: {
    indices: undefined,
    pattern: undefined,
    hasExactMatches: false,
    perPage: 10,
    page: 1,
    sortBy: undefined,
    sortAsc: true,
  }
}

export default handleActions({
  [selectTimeField](state, { payload }) {
    const { timeFields } = payload;

    return {
      ...state,
      timeFields: {
        ...state.timeFields,
        timeFields,
      },
    };
  },
  [fetchedTimeFields](state, { payload }) {
    const { timeFields } = payload;

    return {
      ...state,
      timeFields: {
        ...state.timeFields,
        timeFields,
      },
    };
  },
  [selectTimeField](state, { payload }) {
    const { timeField } = payload;

    return {
      ...state,
      timeFields: {
        ...state.timeFields,
        selectedTimeField: timeField,
      },
    };
  },
  [fetchedIndices](state, { payload }) {
    const { indices, hasExactMatches, pattern } = payload;

    return {
      ...state,
      results: {
        ...state.results,
        indices,
        hasExactMatches,
        pattern,
      },
    };
  },
  [changeSort](state, { payload }) {
    const { sortBy, sortAsc } = payload;

    return {
      ...state,
      results: {
        ...state.results,
        sortBy,
        sortAsc: sortBy === state.results.sortBy ? !state.results.sortAsc: sortAsc,
      },
    };
  },
  [goToNextPage](state, { payload }) {
    return {
      ...state,
      results: {
        ...state.results,
        page: getNextPage(state.results),
      },
    };
  },
  [goToPreviousPage](state, { payload }) {
    return {
      ...state,
      results: {
        ...state.results,
        page: getPreviousPage(state.results),
      },
    };
  },
  [includeSystemIndices](state, action) {
    return {
      ...state,
      isIncludingSystemIndices: true,
    };
  },
  [excludeSystemIndices](state, action) {
    return {
      ...state,
      isIncludingSystemIndices: false,
    };
  },
  [creatingIndexPattern](state, action) {
    return {
      ...state,
      isCreating: true,
    };
  },
  [createdIndexPattern](state, action) {
    return {
      ...state,
      isCreating: false,
    };
  },
}, defaultState);

function getNextPage({ page, perPage, indices }) {
  const pages = Math.ceil(indices.length / perPage);
  const nextPage = page + 1;
  return nextPage > pages
    ? 1
    : nextPage;
}

function getPreviousPage({ page, perPage, indices }) {
  const pages = Math.ceil(indices.length / perPage);
  const previousPage = page - 1;
  return previousPage < 1
    ? pages
    : previousPage;
}

function getFilteredIndices(indices, isIncludingSystemIndices) {
  if (!indices) {
    return indices;
  }

  if (isIncludingSystemIndices) {
    return indices;
  }

  // All system indices begin with a period.
  return indices.filter(index => !index.name.startsWith('.'));
}

function getPaginatedIndices(indices, page, perPage) {
  const pagesOfIndices = chunk(indices, perPage);
  return pagesOfIndices[page - 1];
}

export const getFilteredAndPaginatedIndices = createSelector(
  [
    state => state.results,
    state => state.isIncludingSystemIndices
  ],
  ({
    indices: allIndices,
    page,
    perPage,
    sortBy,
    sortAsc,
  }, isIncludingSystemIndices) => {
    if (allIndices === undefined) {
      return {
        indices: undefined,
        numOfPages: 0,
      };
    }

    let filteredIndices = getFilteredIndices(allIndices, isIncludingSystemIndices);
    if (!!sortBy) {
      filteredIndices = sortByLodash(filteredIndices, sortBy);
      if (!sortAsc) {
        filteredIndices.reverse();
      }
    }

    const numOfPages = Math.ceil(filteredIndices.length / perPage);
    const indices = getPaginatedIndices(filteredIndices, page, perPage);
    return {
      indices,
      numOfPages,
    }
  }
);

export const getPattern = state => getIndexPatternCreate(state).results.pattern;
export const getSelectedTimeField = state => getIndexPatternCreate(state).timeFields.selectedTimeField;
