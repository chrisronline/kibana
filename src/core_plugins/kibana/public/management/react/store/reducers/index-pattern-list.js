import { handleActions } from 'redux-actions';

import {
  fetchedIndexPatterns,
  setTransientTableId,
} from '../actions/index-pattern-list';

import {
  getIndexPatternList,
} from '../../reducers';

const defaultState = {
  indexPatterns: undefined,
  transientTableId: undefined,
};

export default handleActions({
  [fetchedIndexPatterns](state, { payload }) {
    const { indexPatterns } = payload;

    return {
      ...state,
      indexPatterns,
    };
  },
  [setTransientTableId](state, { payload: { id: transientTableId } }) {
    return {
      ...state,
      transientTableId,
    };
  }
}, defaultState);

export const getIndexPatterns = state => getIndexPatternList(state).indexPatterns;
export const getPathToIndexPatterns = () => 'indexPatterns';
