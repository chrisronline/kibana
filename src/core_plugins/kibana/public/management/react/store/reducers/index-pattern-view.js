import { handleActions } from 'redux-actions';

import {
  fetchedIndexPattern,
  setAsDefaultIndexPattern,
} from '../actions/index-pattern-view';

import {
  getIndexPatternView,
} from '../../reducers';

const defaultState = {
  indexPattern: {
    pattern: undefined,
    fields: undefined,
    timeFieldName: undefined,
    isDefault: undefined,
    id: undefined,
  },
  fieldsTable: {},
  tabs: {},
};

export const indexPatternView = handleActions({
  [fetchedIndexPattern](state, { payload: { indexPattern } }) {
    return {
      ...state,
      indexPattern,
    };
  },
  [setAsDefaultIndexPattern](state) {
    return {
      ...state,
      indexPattern: {
        ...state.indexPattern,
        isDefault: true,
      }
    };
  }
}, defaultState);

export const getPathToFields = () => 'indexPattern.fields';
export const getTabs = state => getIndexPatternView(state).tabs;
