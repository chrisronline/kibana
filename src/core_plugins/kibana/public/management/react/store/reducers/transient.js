import { handleActions } from 'redux-actions';

import {
  change,
  createItem,
} from '../actions/transient';

const defaultState = {};

export default handleActions({
  [createItem](state, { payload: { id, defaultData }}) {
    return {
      ...state,
      [id]: defaultData || {},
    }
  },
  [change](state, { payload: { id, data } }) {
    return {
      ...state,
      [id]: {
        ...state[id],
        ...data
      },
    };
  },
}, defaultState);
