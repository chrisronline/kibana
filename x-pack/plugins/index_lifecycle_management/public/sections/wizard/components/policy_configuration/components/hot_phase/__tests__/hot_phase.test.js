/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { HotPhase } from '../';
import { indexLifecycleManagement } from '../../../../../../../store/reducers';
import {
  PHASE_HOT,
  PHASE_ENABLED,
  PHASE_ROLLOVER_ENABLED,
  PHASE_ROLLOVER_MAX_SIZE_STORED,
  PHASE_ROLLOVER_MAX_SIZE_STORED_UNITS,
  PHASE_ROLLOVER_MAX_AGE,
  PHASE_ROLLOVER_MAX_AGE_UNITS
} from '../../../../../../../store/constants';

jest.mock('@elastic/eui', () => ({
  EuiFlexGroup: props => props.children || '',
  EuiFlexItem: props => props.children || '',
  EuiSpacer: props => props.children || '',
  EuiText: props => props.children || '',
  EuiTextColor: props => props.children || '',
  EuiFieldNumber: props => props.children || '',
  EuiSelect: props => props.children || '',
  EuiSwitch: props => props.children || '',
  EuiLink: props => props.children || '',
  EuiFormRow: props => props.children || '',
  EuiDescribedFormGroup: props => props.children || '',
  EuiBetaBadge: props => props.children || '',
}));

jest.mock('../../../../../../../store/actions', () => ({
  setPhaseData: jest.fn(),
}));

const defaultStore = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
};

const defaultState = indexLifecycleManagement({}, {});
const actions = require('../../../../../../../store/actions');

describe('HotPhase', () => {
  it('should render just the activate button if not activated', () => {
    const store = {
      getState: () => ({
        ...defaultState,
      }),
      ...defaultStore,
    };

    const component = mount(
      <Provider store={store}>
        <HotPhase
          validate={() => {}}
          errors={{}}
          isShowingErrors={false}
        />
      </Provider>
    );

    expect(component.find('HotPhase')).toMatchSnapshot();
  });

  it('should show configuration options when activated', async () => {
    const store = {
      getState: () => ({
        ...defaultState,
        policies: {
          ...defaultState.policies,
          selectedPolicy: {
            ...defaultState.policies.selectedPolicy,
            phases: {
              ...defaultState.policies.selectedPolicy.phases,
              [PHASE_HOT]: {
                ...defaultState.policies.selectedPolicy.phases[PHASE_HOT],
                [PHASE_ENABLED]: true
              }
            }
          }
        }
      }),
      ...defaultStore,
    };

    const component = mount(
      <Provider store={store}>
        <HotPhase
          validate={() => {}}
          errors={{}}
          isShowingErrors={false}
        />
      </Provider>
    );

    expect(component.find('HotPhase')).toMatchSnapshot();
  });

  describe('should change fields through actions and validate if necessary', () => {
    const store = {
      getState: () => ({
        ...defaultState,
        policies: {
          ...defaultState.policies,
          selectedPolicy: {
            ...defaultState.policies.selectedPolicy,
            phases: {
              ...defaultState.policies.selectedPolicy.phases,
              [PHASE_HOT]: {
                ...defaultState.policies.selectedPolicy.phases[PHASE_HOT],
                [PHASE_ENABLED]: true
              }
            }
          }
        }
      }),
      ...defaultStore,
    };

    const storeWithRolloverEnabled = {
      getState: () => ({
        ...defaultState,
        policies: {
          ...defaultState.policies,
          selectedPolicy: {
            ...defaultState.policies.selectedPolicy,
            phases: {
              ...defaultState.policies.selectedPolicy.phases,
              [PHASE_HOT]: {
                ...defaultState.policies.selectedPolicy.phases[PHASE_HOT],
                [PHASE_ENABLED]: true,
                [PHASE_ROLLOVER_ENABLED]: true
              }
            }
          }
        }
      }),
      ...defaultStore,
    };

    test('rollover', async () => {
      const validate = jest.fn();
      const component = mount(
        <Provider store={store}>
          <HotPhase
            validate={validate}
            errors={{}}

            isShowingErrors={false}
          />
        </Provider>
      );

      await component.find('EuiSwitch').get(0).props.onChange({ target: { checked: true } });
      expect(validate).toHaveBeenCalled();
      expect(actions.setPhaseData).toHaveBeenCalledWith(PHASE_HOT, PHASE_ROLLOVER_ENABLED, true);
    });

    test('rollover disabled', async () => {
      const validate = jest.fn();
      const component = mount(
        <Provider store={storeWithRolloverEnabled}>
          <HotPhase
            validate={validate}
            errors={{}}

            isShowingErrors={false}
          />
        </Provider>
      );

      await component.find('EuiSwitch').get(0).props.onChange({ target: { checked: false } });
      expect(validate).toHaveBeenCalled();
      expect(actions.setPhaseData).toHaveBeenCalledWith(PHASE_HOT, PHASE_ROLLOVER_ENABLED, false);
    });

    test('max index size', async () => {
      const validate = jest.fn();
      const component = mount(
        <Provider store={storeWithRolloverEnabled}>
          <HotPhase
            validate={validate}
            errors={{}}

            isShowingErrors={false}
          />
        </Provider>
      );

      await component.find('EuiFieldNumber').get(0).props.onChange({ target: { value: '2' } });
      expect(actions.setPhaseData).toHaveBeenCalledWith(PHASE_HOT, PHASE_ROLLOVER_MAX_SIZE_STORED, '2');
    });

    test('max index size units', async () => {
      const validate = jest.fn();
      const component = mount(
        <Provider store={storeWithRolloverEnabled}>
          <HotPhase
            validate={validate}
            errors={{}}

            isShowingErrors={false}
          />
        </Provider>
      );

      await component.find('EuiSelect').get(0).props.onChange({ target: { value: 'gb' } });
      expect(actions.setPhaseData).toHaveBeenCalledWith(PHASE_HOT, PHASE_ROLLOVER_MAX_SIZE_STORED_UNITS, 'gb');
    });

    test('max age', async () => {
      const validate = jest.fn();
      const component = mount(
        <Provider store={storeWithRolloverEnabled}>
          <HotPhase
            validate={validate}
            errors={{}}

            isShowingErrors={false}
          />
        </Provider>
      );

      await component.find('EuiFieldNumber').get(1).props.onChange({ target: { value: '2' } });
      expect(actions.setPhaseData).toHaveBeenCalledWith(PHASE_HOT, PHASE_ROLLOVER_MAX_AGE, '2');
    });

    test('max age units', async () => {
      const validate = jest.fn();
      const component = mount(
        <Provider store={storeWithRolloverEnabled}>
          <HotPhase
            validate={validate}
            errors={{}}

            isShowingErrors={false}
          />
        </Provider>
      );

      await component.find('EuiSelect').get(1).props.onChange({ target: { value: 'h' } });
      expect(actions.setPhaseData).toHaveBeenCalledWith(PHASE_HOT, PHASE_ROLLOVER_MAX_AGE_UNITS, 'h');
    });
  });

  it('should show errors', () => {
    const store = {
      getState: () => ({
        ...defaultState,
        policies: {
          ...defaultState.policies,
          selectedPolicy: {
            ...defaultState.policies.selectedPolicy,
            phases: {
              ...defaultState.policies.selectedPolicy.phases,
              [PHASE_HOT]: {
                ...defaultState.policies.selectedPolicy.phases[PHASE_HOT],
                [PHASE_ENABLED]: true
              }
            }
          }
        }
      }),
      ...defaultStore,
    };

    const component = mount(
      <Provider store={store}>
        <HotPhase
          validate={() => {}}
          errors={{
            [PHASE_ROLLOVER_MAX_SIZE_STORED]: ['Error'],
            [PHASE_ROLLOVER_MAX_SIZE_STORED_UNITS]: ['Error'],
            [PHASE_ROLLOVER_MAX_AGE]: ['Error'],
            [PHASE_ROLLOVER_MAX_AGE_UNITS]: ['Error'],
          }}
          isShowingErrors={true}
        />
      </Provider>
    );

    expect(component.find('HotPhase')).toMatchSnapshot();
  });
});
