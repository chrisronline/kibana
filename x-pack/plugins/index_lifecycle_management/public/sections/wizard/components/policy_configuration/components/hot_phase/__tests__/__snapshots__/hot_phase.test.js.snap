// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotPhase should render just the activate button if not activated 1`] = `
<HotPhase
  errors={Object {}}
  isShowingErrors={false}
  phaseData={
    Object {
      "phaseEnabled": true,
      "rolloverEnabled": true,
      "selectedAlias": "",
      "selectedMaxAge": "",
      "selectedMaxAgeUnits": "s",
      "selectedMaxDocSize": "",
      "selectedMaxSizeStored": "",
      "selectedMaxSizeStoredUnits": "gb",
    }
  }
  setPhaseData={[Function]}
  validate={[Function]}
>
  <EuiDescribedFormGroup
    description={
      <UNDEFINED>
        <p>
          This phase is required. Your index is being queried and actively written to. You can optimize this phase for write throughput.
        </p>
      </UNDEFINED>
    }
    fullWidth={true}
    title={
      <div>
        <span
          className="eui-displayInlineBlock eui-alignMiddle"
        >
          Hot phase
        </span>
         
        <EuiBetaBadge
          className="eui-alignMiddle"
          iconType="check"
          label="Enabled"
        />
      </div>
    }
    titleSize="s"
  >
    <EuiFormRow
      hasEmptyLabelSpace={true}
      helpText={
        <p>
          If true, rollover the index when it gets too big or too old. The alias switches to the new index.
           
          <EuiLink
            href="https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html"
          >
            Learn more
          </EuiLink>
        </p>
      }
    >
      <EuiSwitch
        checked={true}
        label="Enable rollover"
        onChange={[Function]}
      />
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiFlexGroup>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxSizeStored"
          errors={Object {}}
          isShowingErrors={false}
          label="Maximum index size"
        >
          <EuiFormRow
            isInvalid={false}
            label="Maximum index size"
          >
            <EuiFieldNumber
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              value=""
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxSizeStoredUnits"
          errors={Object {}}
          hasEmptyLabelSpace={true}
          isShowingErrors={false}
        >
          <EuiFormRow
            hasEmptyLabelSpace={true}
            isInvalid={false}
          >
            <EuiSelect
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "gigabytes",
                    "value": "gb",
                  },
                  Object {
                    "text": "documents",
                    "value": "d",
                  },
                ]
              }
              value="gb"
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
    </EuiFlexGroup>
    <EuiSpacer />
    <EuiFlexGroup>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxAge"
          errors={Object {}}
          isShowingErrors={false}
          label="Maximum age"
        >
          <EuiFormRow
            isInvalid={false}
            label="Maximum age"
          >
            <EuiFieldNumber
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              value=""
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxAgeUnits"
          errors={Object {}}
          hasEmptyLabelSpace={true}
          isShowingErrors={false}
        >
          <EuiFormRow
            hasEmptyLabelSpace={true}
            isInvalid={false}
          >
            <EuiSelect
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "days",
                    "value": "d",
                  },
                  Object {
                    "text": "hours",
                    "value": "h",
                  },
                ]
              }
              value="s"
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiDescribedFormGroup>
</HotPhase>
`;

exports[`HotPhase should show configuration options when activated 1`] = `
<HotPhase
  errors={Object {}}
  isShowingErrors={false}
  phaseData={
    Object {
      "phaseEnabled": true,
      "rolloverEnabled": true,
      "selectedAlias": "",
      "selectedMaxAge": "",
      "selectedMaxAgeUnits": "s",
      "selectedMaxDocSize": "",
      "selectedMaxSizeStored": "",
      "selectedMaxSizeStoredUnits": "gb",
    }
  }
  setPhaseData={[Function]}
  validate={[Function]}
>
  <EuiDescribedFormGroup
    description={
      <UNDEFINED>
        <p>
          This phase is required. Your index is being queried and actively written to. You can optimize this phase for write throughput.
        </p>
      </UNDEFINED>
    }
    fullWidth={true}
    title={
      <div>
        <span
          className="eui-displayInlineBlock eui-alignMiddle"
        >
          Hot phase
        </span>
         
        <EuiBetaBadge
          className="eui-alignMiddle"
          iconType="check"
          label="Enabled"
        />
      </div>
    }
    titleSize="s"
  >
    <EuiFormRow
      hasEmptyLabelSpace={true}
      helpText={
        <p>
          If true, rollover the index when it gets too big or too old. The alias switches to the new index.
           
          <EuiLink
            href="https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html"
          >
            Learn more
          </EuiLink>
        </p>
      }
    >
      <EuiSwitch
        checked={true}
        label="Enable rollover"
        onChange={[Function]}
      />
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiFlexGroup>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxSizeStored"
          errors={Object {}}
          isShowingErrors={false}
          label="Maximum index size"
        >
          <EuiFormRow
            isInvalid={false}
            label="Maximum index size"
          >
            <EuiFieldNumber
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              value=""
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxSizeStoredUnits"
          errors={Object {}}
          hasEmptyLabelSpace={true}
          isShowingErrors={false}
        >
          <EuiFormRow
            hasEmptyLabelSpace={true}
            isInvalid={false}
          >
            <EuiSelect
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "gigabytes",
                    "value": "gb",
                  },
                  Object {
                    "text": "documents",
                    "value": "d",
                  },
                ]
              }
              value="gb"
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
    </EuiFlexGroup>
    <EuiSpacer />
    <EuiFlexGroup>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxAge"
          errors={Object {}}
          isShowingErrors={false}
          label="Maximum age"
        >
          <EuiFormRow
            isInvalid={false}
            label="Maximum age"
          >
            <EuiFieldNumber
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              value=""
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxAgeUnits"
          errors={Object {}}
          hasEmptyLabelSpace={true}
          isShowingErrors={false}
        >
          <EuiFormRow
            hasEmptyLabelSpace={true}
            isInvalid={false}
          >
            <EuiSelect
              isInvalid={false}
              key=".0"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "days",
                    "value": "d",
                  },
                  Object {
                    "text": "hours",
                    "value": "h",
                  },
                ]
              }
              value="s"
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiDescribedFormGroup>
</HotPhase>
`;

exports[`HotPhase should show errors 1`] = `
<HotPhase
  errors={
    Object {
      "selectedMaxAge": Array [
        "Error",
      ],
      "selectedMaxAgeUnits": Array [
        "Error",
      ],
      "selectedMaxSizeStored": Array [
        "Error",
      ],
      "selectedMaxSizeStoredUnits": Array [
        "Error",
      ],
    }
  }
  isShowingErrors={true}
  phaseData={
    Object {
      "phaseEnabled": true,
      "rolloverEnabled": true,
      "selectedAlias": "",
      "selectedMaxAge": "",
      "selectedMaxAgeUnits": "s",
      "selectedMaxDocSize": "",
      "selectedMaxSizeStored": "",
      "selectedMaxSizeStoredUnits": "gb",
    }
  }
  setPhaseData={[Function]}
  validate={[Function]}
>
  <EuiDescribedFormGroup
    description={
      <UNDEFINED>
        <p>
          This phase is required. Your index is being queried and actively written to. You can optimize this phase for write throughput.
        </p>
        <EuiTextColor
          color="danger"
        >
          <EuiText>
            <p>
              This phase contains errors
            </p>
          </EuiText>
        </EuiTextColor>
      </UNDEFINED>
    }
    fullWidth={true}
    title={
      <div>
        <span
          className="eui-displayInlineBlock eui-alignMiddle"
        >
          Hot phase
        </span>
         
        <EuiBetaBadge
          className="eui-alignMiddle"
          iconType="check"
          label="Enabled"
        />
      </div>
    }
    titleSize="s"
  >
    <EuiFormRow
      hasEmptyLabelSpace={true}
      helpText={
        <p>
          If true, rollover the index when it gets too big or too old. The alias switches to the new index.
           
          <EuiLink
            href="https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html"
          >
            Learn more
          </EuiLink>
        </p>
      }
    >
      <EuiSwitch
        checked={true}
        label="Enable rollover"
        onChange={[Function]}
      />
    </EuiFormRow>
    <EuiSpacer
      size="m"
    />
    <EuiFlexGroup>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxSizeStored"
          errors={
            Object {
              "selectedMaxAge": Array [
                "Error",
              ],
              "selectedMaxAgeUnits": Array [
                "Error",
              ],
              "selectedMaxSizeStored": Array [
                "Error",
              ],
              "selectedMaxSizeStoredUnits": Array [
                "Error",
              ],
            }
          }
          isShowingErrors={true}
          label="Maximum index size"
        >
          <EuiFormRow
            error={
              Array [
                "Error",
              ]
            }
            isInvalid={true}
            label="Maximum index size"
          >
            <EuiFieldNumber
              isInvalid={true}
              key=".0"
              onChange={[Function]}
              value=""
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxSizeStoredUnits"
          errors={
            Object {
              "selectedMaxAge": Array [
                "Error",
              ],
              "selectedMaxAgeUnits": Array [
                "Error",
              ],
              "selectedMaxSizeStored": Array [
                "Error",
              ],
              "selectedMaxSizeStoredUnits": Array [
                "Error",
              ],
            }
          }
          hasEmptyLabelSpace={true}
          isShowingErrors={true}
        >
          <EuiFormRow
            error={
              Array [
                "Error",
              ]
            }
            hasEmptyLabelSpace={true}
            isInvalid={true}
          >
            <EuiSelect
              isInvalid={true}
              key=".0"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "gigabytes",
                    "value": "gb",
                  },
                  Object {
                    "text": "documents",
                    "value": "d",
                  },
                ]
              }
              value="gb"
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
    </EuiFlexGroup>
    <EuiSpacer />
    <EuiFlexGroup>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxAge"
          errors={
            Object {
              "selectedMaxAge": Array [
                "Error",
              ],
              "selectedMaxAgeUnits": Array [
                "Error",
              ],
              "selectedMaxSizeStored": Array [
                "Error",
              ],
              "selectedMaxSizeStoredUnits": Array [
                "Error",
              ],
            }
          }
          isShowingErrors={true}
          label="Maximum age"
        >
          <EuiFormRow
            error={
              Array [
                "Error",
              ]
            }
            isInvalid={true}
            label="Maximum age"
          >
            <EuiFieldNumber
              isInvalid={true}
              key=".0"
              onChange={[Function]}
              value=""
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
      <EuiFlexItem
        style={
          Object {
            "maxWidth": 188,
          }
        }
      >
        <ErrableFormRow
          errorKey="selectedMaxAgeUnits"
          errors={
            Object {
              "selectedMaxAge": Array [
                "Error",
              ],
              "selectedMaxAgeUnits": Array [
                "Error",
              ],
              "selectedMaxSizeStored": Array [
                "Error",
              ],
              "selectedMaxSizeStoredUnits": Array [
                "Error",
              ],
            }
          }
          hasEmptyLabelSpace={true}
          isShowingErrors={true}
        >
          <EuiFormRow
            error={
              Array [
                "Error",
              ]
            }
            hasEmptyLabelSpace={true}
            isInvalid={true}
          >
            <EuiSelect
              isInvalid={true}
              key=".0"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": "days",
                    "value": "d",
                  },
                  Object {
                    "text": "hours",
                    "value": "h",
                  },
                ]
              }
              value="s"
            />
          </EuiFormRow>
        </ErrableFormRow>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiDescribedFormGroup>
</HotPhase>
`;
